src/providers/logout/logout.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { AlertController, App } from 'ionic-angular';
+import { Storage } from '@ionic/storage';
+import { LoginPage } from '../../pages/login/login';
+
+/*
+  Generated class for the LogoutProvider provider.
+
+  See https://angular.io/guide/dependency-injection for more info on providers
+  and Angular DI.
+*/
+@Injectable()
+export class LogoutProvider {
+
+  constructor( private storage: Storage, private app: App, private alertCtrl: AlertController) {
+    console.log('Hello LogoutProvider Provider');
+  }
+
+  showLogout() {
+    // Handle logout action sheet button
+    const logout = () => {
+      return this.storage.ready()
+      .then(() => {
+        Promise.all([
+        this.storage.remove("accessKey"),
+        this.storage.remove("accessExpiry"),
+        this.storage.remove("refreshKey"),
+        this.storage.remove("refreshExpiry"),
+        this.storage.remove("username")])})
+    }
+
+    let alert = this.alertCtrl.create({
+      title: 'Log out',
+      message: 'Log out of Iris?',
+      buttons: [
+        {
+          text: 'Cancel',
+          role: 'cancel'
+        },
+        {
+          text: 'Log out',
+          handler: () => {
+            // Logic again needed for transitions to work properly.
+            // Dismiss alert, then navigate to login page.
+            let navTransition = alert.dismiss()
+            logout().then(() => {
+              navTransition.then(() => {
+                this.app.getRootNav().setRoot(LoginPage, {loggedOut: true});
```

Comment:
Seems like Ionic idioms dictate that navigation should be handled by the pages, rather than a provider. I think `getRootNav` and `getActiveNav` are generally discouraged. Instead, return a promise based on the `navTransition` that the pages can use to navigate back to LoginPage, using their own NavControllers